portal-impl/src/com/liferay/portal/security/access/control/AccessControlImpl.java

```diff
@@ -108,8 +108,10 @@ public AuthVerifierResult.State verifyRequest() throws PortalException {
 			authVerifierResult = authVerifierPipeline.verifyRequest(
 				accessControlContext);
 
-			if (authVerifierResult.getState() !=
-					AuthVerifierResult.State.SUCCESS) {
+			if ((authVerifierResult.getState() !=
+					AuthVerifierResult.State.SUCCESS) &&
+				(authVerifierResult.getState() !=
```

Comment:
Why exclude INVALID_CREDENTIALS here?

Reply:
@arthurchan35  Because otherwise it will try the "portal" AuthVerifierPipeline, which might return SUCCESS. INVALID_CREDENTIALS is special in that the pipeline processing should terminate immediately.